src/instruction/instructions/load/ldr.ts

```diff
@@ -125,34 +148,79 @@ export class LdrRegisterOffsetInstruction extends BaseInstruction {
 }
 
 /**
- * Represents a 'LOAD' instruction - LDR (pointer + offset) - word
+ * Represents a 'LOAD' instruction - LDR (programm counter + offset) - word
  */
 export class LdrRegisterInstruction extends BaseInstruction {
   public name: string = 'LDR'
   public pattern: string = '01001XXXXXXXXXXX'
   private immPattern: string = '01001000XXXXXXXX'
   private rtPattern: string = '01001XXX00000000'
-  private expectedOptionCount: number = 3
+  private expectedOptionCountMin: number = 2
+  private expectedOptionCountMax: number = 3
+  public needsLabels: boolean = true
 
   public canEncodeInstruction(name: string, options: string[]): boolean {
     return (
       super.canEncodeInstruction(name, options) &&
-      isOptionCountValid(options, this.expectedOptionCount) &&
-      isImmediate(options[2]) &&
-      registerStringHasBrackets(options[1], options[2]) &&
-      'PC' === removeBracketsFromRegisterString(options[1])
+      isOptionCountValid(
+        options,
+        this.expectedOptionCountMin,
+        this.expectedOptionCountMax
+      ) &&
+      (LdrRegisterInstruction.isLabelOffsetInstruction(options) ||
+        (isPCRegister(options[1]) &&
+          (options.length == this.expectedOptionCountMin ||
+            isImmediate(options[2]))))
     )
   }
 
-  public encodeInstruction(options: string[]): Halfword[] {
-    checkOptionCount(options, 3)
+  public encodeInstruction(
+    options: string[],
+    labels?: ILabelOffsets
+  ): Halfword[] {
+    checkOptionCount(
+      options,
+      this.expectedOptionCountMin,
+      this.expectedOptionCountMax
+    )
+    if (!LdrRegisterInstruction.isLabelOffsetInstruction(options)) {
+      checkBracketsOnLastOptions(
+        options,
+        this.expectedOptionCountMin,
+        this.expectedOptionCountMax
+      )
+      if (!isPCRegister(options[1])) {
+        throw new InstructionError('Second param is not PC register')
+      }
```

Comment:
`isLiteralString` just checks wheter provided option is a valid register or starts or ends with brackets ([,])
since labels should not be register string nor include any brackets, I would not add this hint here (all other strings are accepted as label).

Reply:
I am not a 100% sure if I got you right, and I was not precise, unfortunately. I meant the user could be informed that it is not a pseudo instruction (instead of "not a label") since isLabelOffsetInstruction does that check, right? I did not find isLiteralString in the code section that I marked ... It is a minor issue though that is not worthy of too much reflection =)